anima/env/mayaEnv/publish.py

```diff
@@ -933,6 +969,61 @@ def check_empty_shapes(progress_controller=None):
         )
 
 
+@publisher('model')
+def check_default_uv_set(progress_controller=None):
+    """All mesh have default UVSet map1
+
+    checks if all mesh have 1 default UVSet named as map1
+    """
+    if progress_controller is None:
+        progress_controller = ProgressControllerBase()
+
+    # skip if this is a representation
+    v = staging.get('version')
+    if v and Representation.repr_separator in v.take_name:
+        progress_controller.complete()
+        return
+
+    all_meshes = pm.ls(type='mesh')
+    progress_controller.maximum = len(all_meshes)
+    nodes_with_non_default_uvset = []
+    for node in all_meshes:
+        if len(node.getUVSetNames()) != 1 or node.getUVSetNames()[0] != 'map1':
+            nodes_with_non_default_uvset.append(node)
+        progress_controller.increment()
+
+    progress_controller.complete()
+    if len(nodes_with_non_default_uvset) > 0:
+        # get transform nodes
+        tra_nodes = map(
+            lambda x: x.getParent(),
+            nodes_with_non_default_uvset
+        )
+        pm.select(tra_nodes)
+        raise RuntimeError(
+            """There are nodes with <b>non default UVSet (map1)</b>:
+            <br><br>%s""" %
+            '<br>'.join(
+                map(lambda x: x.name(),
+                    tra_nodes[:MAX_NODE_DISPLAY])
+            )
+        )
+
+
+def check_default_uv_set___fix():
```

Comment:
as per my previous comment you should delete this code. It should be free to use more than one UV set.

Reply:
after changing the code above as you mentioned... I think we can still use this fix_function because it only renames the uvset as map1 if only 1 uvset exists and if it's name is not map1.
I am also planning to check if map1 is empty or not. Because in most cases when we transfer alembics or objs from various software... maya tends to create an empty default uvset as map1. So the actual uvs stay in the other set where map1 is empty. So what we can do for this fix_function is to check if there is 2 uvsets and if map1 is empty we can copy the uvs to map1 and delete the other one maybe.
what do you say?

Reply:
Yeah I think that is plausable